app/controllers/users_controller.rb

```diff
@@ -18,9 +25,48 @@ def create
     end
   end
 
+  def edit; end
+
+  def update
+    if @user.update_attributes user_params
+      flash[:success] = t "controllers.user.m_suc_update"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @user.destroy
+    flash[:success] = t "controllers.user.m_suc_delete"
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user).permit User::ATTRIBUTE_PARAMS
   end
+
+  def logged_in_user
+    return if logged_in?
+    store_location
```

Comment:
hàm này xử lý những gì? nói qua cho t xem nào

Reply:
Khi vào các action cần đăng nhập, nó sẽ gọi tới hàm trên để kiểm tra xem đã đăng nhập chưa. Hàm này sẽ kiểm tra user đã được log in chưa, nếu chưa login thì nó sẽ lưu cái đường dẫn hiện tại vào session, chuyển qua bước login, sau khi login xong, nếu trong ss có tồn tại liên kết thì nó chuyển hướng tới đó, không thì nó chuyển hướng tới liên kết mặc định ( hiện trang user)